### PR TITLE
Replacing the bluetoothmessage tuples with structs

### PR DESCRIPTION
**Pull request by zakorgy**

### CODE REVIEW

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -365,23 +365,25 @@ impl BluetoothManager {
...
-                                      Ok(p) => Some(p as i8),
-                                      Err(_) => None,
-                                  },
-                                  match device.get_rssi() {
-                                      Ok(p) => Some(p as i8),
-                                      Err(_) => None,
-                                  },
-                ));
+                let message = Ok(
+                                BluetoothDeviceMsg {
```
Comment (dati91) [inline components/net/bluetooth_thread.rs]: move this one line above

File: components/net/bluetooth_thread.rs [hunk 2]
```diff
@@ -365,23 +365,25 @@ impl BluetoothManager {
...
+                                    id: address,
+                                    name: device.get_name().ok(),
+                                    device_class: device.get_class().ok(),
+                                    vendor_id_source: device.get_vendor_id_source().ok(),
+                                    vendor_id: device.get_vendor_id().ok(),
+                                    product_id: device.get_product_id().ok(),
+                                    product_version: device.get_device_id().ok(),
+                                    appearance: device.get_appearance().ok(),
+                                    tx_power: match device.get_tx_power() {
```
Comment (dati91) [inline components/net/bluetooth_thread.rs]: can we `map` here as well? or just the `Option` can be mapped?

File: components/net/bluetooth_thread.rs [hunk 3]
```diff
@@ -443,7 +445,12 @@ impl BluetoothManager {
         for service in services {
             if service.is_primary().unwrap_or(false) {
                 if let Ok(uuid) = service.get_uuid() {
-                    return drop(sender.send(Ok((uuid, true, service.get_object_path()))));
+                    return drop(sender.send(Ok(
```
Comment (dati91) [inline components/net/bluetooth_thread.rs]: move one line up

File: components/net/bluetooth_thread.rs (hunk 2 above)
Comment (zakorgy) [inline components/net/bluetooth_thread.rs]: we can use map this way `tx_power: device.get_tx_power().ok().map(|p| p as i8),`

**Comment by dati91** [comment]
lgtm

File: components/net/bluetooth_thread.rs [hunk 4]
```diff
@@ -443,7 +438,11 @@ impl BluetoothManager {
         for service in services {
             if service.is_primary().unwrap_or(false) {
                 if let Ok(uuid) = service.get_uuid() {
-                    return drop(sender.send(Ok((uuid, true, service.get_object_path()))));
+                    return drop(sender.send(Ok(BluetoothServiceMsg {
+                                                   uuid: uuid,
+                                                   is_primary: true,
+                                                   instance_id: service.get_object_path(),
+                                               })));
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:445]: iirc, the style says that the `)`s should be on the same line

File: components/net/bluetooth_thread.rs [hunk 5]
```diff
@@ -589,13 +600,15 @@ impl BluetoothManager {
...
+                                         uuid: uuid,
+                                         instance_id: descriptor.get_object_path(),
+                                     });
             }
         }
         if descriptors_vec.is_empty() {
             return drop(sender.send(Err(String::from(DESCRIPTOR_ERROR))));
         }
-
         let _ = sender.send(Ok(descriptors_vec));
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:612]: why `let _`?

File: components/script/dom/bluetooth.rs [hunk 6]
```diff
@@ -137,34 +137,25 @@ impl BluetoothMethods for Bluetooth {
...
-                rssi)) => {
+            Ok(device) => {
                 let ad_data = BluetoothAdvertisingData::new(self.global().r(),
-                                                             appearance,
-                                                             tx_power,
-                                                             rssi);
-                let vendor_id_source = vendor_id_source.map(|vid| match vid.as_str() {
+                                                            device.appearance,
+                                                            device.tx_power,
+                                                            device.rssi);
```
Comment (mmatyas) [inline components/script/dom/bluetooth.rs:144]: maybe indent one level

File: components/script/dom/bluetoothremotegattcharacteristic.rs [hunk 7]
```diff
@@ -127,12 +127,11 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
...
-                                      BluetoothRemoteGATTDescriptor::new(self.global().r(),
-                                                                         self,
-                                                                         DOMString::from(uuid),
-                                                                         instance_id))
-                                  .collect())
+                                 .map(|desc| BluetoothRemoteGATTDescriptor::new(self.global().r(),
+                                                                                self,
+                                                                                DOMString::from(desc.uuid),
+                                                                                desc.instance_id))
+                                 .collect())
```
Comment (mmatyas) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:134]: same line maybe? check the style guide

**Pull request merged**